Bottom: 073207cf2269dd797942b6393b4e95afcd3fc741
Top:    dbc1fac544316dcdd22b47f0c215e143ffedc3f0
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-12-25 11:58:09 +0800

Refresh of tun-tap-use-ptr_ring-instead

---

diff --git a/drivers/vhost/net.c b/drivers/vhost/net.c
index c7fdfe5..c316555 100644
--- a/drivers/vhost/net.c
+++ b/drivers/vhost/net.c
@@ -779,16 +779,6 @@ static void handle_rx(struct vhost_net *net)
 		/* On error, stop handling until the next kick. */
 		if (unlikely(headcount < 0))
 			goto out;
-		if (nvq->rx_ring)
-			msg.msg_control = vhost_net_buf_consume(&nvq->rxq);
-		/* On overrun, truncate and discard */
-		if (unlikely(headcount > UIO_MAXIOV)) {
-			iov_iter_init(&msg.msg_iter, READ, vq->iov, 1, 1);
-			err = sock->ops->recvmsg(sock, &msg,
-						 1, MSG_DONTWAIT | MSG_TRUNC);
-			pr_debug("Discarded rx packet: len %zd\n", sock_len);
-			continue;
-		}
 		/* OK, now we need to know about added descriptors. */
 		if (!headcount) {
 			if (unlikely(vhost_enable_notify(&net->dev, vq))) {
